File: app/src/main/java/com/canli/oya/traininventoryroom/adapters/BindingAdapters.java
package com.canli.oya.traininventoryroom.adapters;

import android.databinding.BindingAdapter;
import android.widget.ImageView;

import com.canli.oya.traininventoryroom.R;
import com.canli.oya.traininventoryroom.utils.GlideApp;

public class BindingAdapters {

    @BindingAdapter("android:src")
    public static void setImageUrl(ImageView view, String url) {
        GlideApp.with(view.getContext())
                .load(url)
                .centerCrop()
                .placeholder(R.drawable.ic_gallery)
                .into(view);
    }
}

File: app/src/main/java/com/canli/oya/traininventoryroom/adapters/BindingAdapters.kt
package com.canli.oya.traininventoryroom.adapters

import android.graphics.drawable.Drawable
import android.view.View
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.canli.oya.traininventoryroom.utils.GlideApp


@BindingAdapter("imageUrl", "placeholder")
fun setImageUrl(view: ImageView, url: String?, placeHolder: Drawable) {
    GlideApp.with(view.context)
            .load(url)
            .centerCrop()
            .placeholder(placeHolder)
            .into(view)
}

@BindingAdapter("visible")
fun View.setVisible(visible: Boolean) {
    visibility = if (visible) View.VISIBLE else View.GONE
}
